// Code generated by protoc-gen-validate. DO NOT EDIT.
// source: v1/authservice.proto

package authPBV1

import (
	"bytes"
	"errors"
	"fmt"
	"net"
	"net/mail"
	"net/url"
	"regexp"
	"sort"
	"strings"
	"time"
	"unicode/utf8"

	"google.golang.org/protobuf/types/known/anypb"
)

// ensure the imports are used
var (
	_ = bytes.MinRead
	_ = errors.New("")
	_ = fmt.Print
	_ = utf8.UTFMax
	_ = (*regexp.Regexp)(nil)
	_ = (*strings.Reader)(nil)
	_ = net.IPv4len
	_ = time.Duration(0)
	_ = (*url.URL)(nil)
	_ = (*mail.Address)(nil)
	_ = anypb.Any{}
	_ = sort.Sort
)

// Validate checks the field values on RegisterAuthRequest with the rules
// defined in the proto definition for this message. If any rules are
// violated, the first error encountered is returned, or nil if there are no violations.
func (m *RegisterAuthRequest) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on RegisterAuthRequest with the rules
// defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// RegisterAuthRequestMultiError, or nil if none found.
func (m *RegisterAuthRequest) ValidateAll() error {
	return m.validate(true)
}

func (m *RegisterAuthRequest) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	if l := utf8.RuneCountInString(m.GetAccessToken()); l < 1 || l > 255 {
		err := RegisterAuthRequestValidationError{
			field:  "AccessToken",
			reason: "value length must be between 1 and 255 runes, inclusive",
		}
		if !all {
			return err
		}
		errors = append(errors, err)
	}

	if m.GetDuration() <= 1 {
		err := RegisterAuthRequestValidationError{
			field:  "Duration",
			reason: "value must be greater than 1",
		}
		if !all {
			return err
		}
		errors = append(errors, err)
	}

	if len(errors) > 0 {
		return RegisterAuthRequestMultiError(errors)
	}

	return nil
}

// RegisterAuthRequestMultiError is an error wrapping multiple validation
// errors returned by RegisterAuthRequest.ValidateAll() if the designated
// constraints aren't met.
type RegisterAuthRequestMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m RegisterAuthRequestMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m RegisterAuthRequestMultiError) AllErrors() []error { return m }

// RegisterAuthRequestValidationError is the validation error returned by
// RegisterAuthRequest.Validate if the designated constraints aren't met.
type RegisterAuthRequestValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e RegisterAuthRequestValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e RegisterAuthRequestValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e RegisterAuthRequestValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e RegisterAuthRequestValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e RegisterAuthRequestValidationError) ErrorName() string {
	return "RegisterAuthRequestValidationError"
}

// Error satisfies the builtin error interface
func (e RegisterAuthRequestValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sRegisterAuthRequest.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = RegisterAuthRequestValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = RegisterAuthRequestValidationError{}

// Validate checks the field values on GetAuthRequest with the rules defined in
// the proto definition for this message. If any rules are violated, the first
// error encountered is returned, or nil if there are no violations.
func (m *GetAuthRequest) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on GetAuthRequest with the rules defined
// in the proto definition for this message. If any rules are violated, the
// result is a list of violation errors wrapped in GetAuthRequestMultiError,
// or nil if none found.
func (m *GetAuthRequest) ValidateAll() error {
	return m.validate(true)
}

func (m *GetAuthRequest) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	if l := utf8.RuneCountInString(m.GetAccessToken()); l < 1 || l > 255 {
		err := GetAuthRequestValidationError{
			field:  "AccessToken",
			reason: "value length must be between 1 and 255 runes, inclusive",
		}
		if !all {
			return err
		}
		errors = append(errors, err)
	}

	if m.GetDuration() <= 1 {
		err := GetAuthRequestValidationError{
			field:  "Duration",
			reason: "value must be greater than 1",
		}
		if !all {
			return err
		}
		errors = append(errors, err)
	}

	if len(errors) > 0 {
		return GetAuthRequestMultiError(errors)
	}

	return nil
}

// GetAuthRequestMultiError is an error wrapping multiple validation errors
// returned by GetAuthRequest.ValidateAll() if the designated constraints
// aren't met.
type GetAuthRequestMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m GetAuthRequestMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m GetAuthRequestMultiError) AllErrors() []error { return m }

// GetAuthRequestValidationError is the validation error returned by
// GetAuthRequest.Validate if the designated constraints aren't met.
type GetAuthRequestValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e GetAuthRequestValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e GetAuthRequestValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e GetAuthRequestValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e GetAuthRequestValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e GetAuthRequestValidationError) ErrorName() string { return "GetAuthRequestValidationError" }

// Error satisfies the builtin error interface
func (e GetAuthRequestValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sGetAuthRequest.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = GetAuthRequestValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = GetAuthRequestValidationError{}

// Validate checks the field values on DestroyAuthRequest with the rules
// defined in the proto definition for this message. If any rules are
// violated, the first error encountered is returned, or nil if there are no violations.
func (m *DestroyAuthRequest) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on DestroyAuthRequest with the rules
// defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// DestroyAuthRequestMultiError, or nil if none found.
func (m *DestroyAuthRequest) ValidateAll() error {
	return m.validate(true)
}

func (m *DestroyAuthRequest) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	if l := utf8.RuneCountInString(m.GetAccessToken()); l < 1 || l > 255 {
		err := DestroyAuthRequestValidationError{
			field:  "AccessToken",
			reason: "value length must be between 1 and 255 runes, inclusive",
		}
		if !all {
			return err
		}
		errors = append(errors, err)
	}

	if len(errors) > 0 {
		return DestroyAuthRequestMultiError(errors)
	}

	return nil
}

// DestroyAuthRequestMultiError is an error wrapping multiple validation errors
// returned by DestroyAuthRequest.ValidateAll() if the designated constraints
// aren't met.
type DestroyAuthRequestMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m DestroyAuthRequestMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m DestroyAuthRequestMultiError) AllErrors() []error { return m }

// DestroyAuthRequestValidationError is the validation error returned by
// DestroyAuthRequest.Validate if the designated constraints aren't met.
type DestroyAuthRequestValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e DestroyAuthRequestValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e DestroyAuthRequestValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e DestroyAuthRequestValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e DestroyAuthRequestValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e DestroyAuthRequestValidationError) ErrorName() string {
	return "DestroyAuthRequestValidationError"
}

// Error satisfies the builtin error interface
func (e DestroyAuthRequestValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sDestroyAuthRequest.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = DestroyAuthRequestValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = DestroyAuthRequestValidationError{}

// Validate checks the field values on IsApiWhiteListRequest with the rules
// defined in the proto definition for this message. If any rules are
// violated, the first error encountered is returned, or nil if there are no violations.
func (m *IsApiWhiteListRequest) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on IsApiWhiteListRequest with the rules
// defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// IsApiWhiteListRequestMultiError, or nil if none found.
func (m *IsApiWhiteListRequest) ValidateAll() error {
	return m.validate(true)
}

func (m *IsApiWhiteListRequest) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	if l := utf8.RuneCountInString(m.GetFullMethodName()); l < 1 || l > 255 {
		err := IsApiWhiteListRequestValidationError{
			field:  "FullMethodName",
			reason: "value length must be between 1 and 255 runes, inclusive",
		}
		if !all {
			return err
		}
		errors = append(errors, err)
	}

	if len(errors) > 0 {
		return IsApiWhiteListRequestMultiError(errors)
	}

	return nil
}

// IsApiWhiteListRequestMultiError is an error wrapping multiple validation
// errors returned by IsApiWhiteListRequest.ValidateAll() if the designated
// constraints aren't met.
type IsApiWhiteListRequestMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m IsApiWhiteListRequestMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m IsApiWhiteListRequestMultiError) AllErrors() []error { return m }

// IsApiWhiteListRequestValidationError is the validation error returned by
// IsApiWhiteListRequest.Validate if the designated constraints aren't met.
type IsApiWhiteListRequestValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e IsApiWhiteListRequestValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e IsApiWhiteListRequestValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e IsApiWhiteListRequestValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e IsApiWhiteListRequestValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e IsApiWhiteListRequestValidationError) ErrorName() string {
	return "IsApiWhiteListRequestValidationError"
}

// Error satisfies the builtin error interface
func (e IsApiWhiteListRequestValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sIsApiWhiteListRequest.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = IsApiWhiteListRequestValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = IsApiWhiteListRequestValidationError{}

// Validate checks the field values on Response with the rules defined in the
// proto definition for this message. If any rules are violated, the first
// error encountered is returned, or nil if there are no violations.
func (m *Response) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on Response with the rules defined in
// the proto definition for this message. If any rules are violated, the
// result is a list of violation errors wrapped in ResponseMultiError, or nil
// if none found.
func (m *Response) ValidateAll() error {
	return m.validate(true)
}

func (m *Response) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	// no validation rules for Code

	// no validation rules for Msg

	if all {
		switch v := interface{}(m.GetProtoAnyData()).(type) {
		case interface{ ValidateAll() error }:
			if err := v.ValidateAll(); err != nil {
				errors = append(errors, ResponseValidationError{
					field:  "ProtoAnyData",
					reason: "embedded message failed validation",
					cause:  err,
				})
			}
		case interface{ Validate() error }:
			if err := v.Validate(); err != nil {
				errors = append(errors, ResponseValidationError{
					field:  "ProtoAnyData",
					reason: "embedded message failed validation",
					cause:  err,
				})
			}
		}
	} else if v, ok := interface{}(m.GetProtoAnyData()).(interface{ Validate() error }); ok {
		if err := v.Validate(); err != nil {
			return ResponseValidationError{
				field:  "ProtoAnyData",
				reason: "embedded message failed validation",
				cause:  err,
			}
		}
	}

	if len(errors) > 0 {
		return ResponseMultiError(errors)
	}

	return nil
}

// ResponseMultiError is an error wrapping multiple validation errors returned
// by Response.ValidateAll() if the designated constraints aren't met.
type ResponseMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m ResponseMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m ResponseMultiError) AllErrors() []error { return m }

// ResponseValidationError is the validation error returned by
// Response.Validate if the designated constraints aren't met.
type ResponseValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e ResponseValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e ResponseValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e ResponseValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e ResponseValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e ResponseValidationError) ErrorName() string { return "ResponseValidationError" }

// Error satisfies the builtin error interface
func (e ResponseValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sResponse.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = ResponseValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = ResponseValidationError{}
